frame/contracts/src/migration/v14.rs

```diff
@@ -0,0 +1,272 @@
+pub fn store_dummy_code<T: Config, OldCurrency>(account: T::AccountId)
+where
+	T: Config,
+	OldCurrency: ReservableCurrency<<T as frame_system::Config>::AccountId> + 'static,
+{
+	use sp_runtime::traits::Hash;
+	use sp_std::vec;
+
+	let len = T::MaxCodeLen::get();
+	let code = vec![42u8; len as usize];
+	let hash = T::Hashing::hash(&code);
+
+	let info = old::CodeInfo {
+		owner: account,
+		deposit: 10_000u32.into(),
+		refcount: u64::MAX,
+		determinism: Determinism::Enforced,
+		code_len: len,
+	};
+	old::CodeInfoOf::<T, OldCurrency>::insert(hash, info);
+}
+
+#[cfg(feature = "try-runtime")]
+#[derive(Encode, Decode)]
+/// Accounts for the balance allocation of a code owner.
+struct BalanceAllocation<T, OldCurrency>
+where
+	T: Config,
+	OldCurrency: ReservableCurrency<<T as frame_system::Config>::AccountId>,
+{
+	/// Total reserved balance as storage deposit for the owner.
+	reserved: old::BalanceOf<T, OldCurrency>,
+	/// Total balance of the owner.
+	total: old::BalanceOf<T, OldCurrency>,
+}
+
+#[derive(Encode, Decode, MaxEncodedLen, DefaultNoBound)]
+pub struct Migration<T, OldCurrency>
+where
+	T: Config,
+	OldCurrency: ReservableCurrency<<T as frame_system::Config>::AccountId>,
+{
+	last_code_hash: Option<CodeHash<T>>,
+	_phantom: PhantomData<(T, OldCurrency)>,
+}
+
+impl<T, OldCurrency> MigrationStep for Migration<T, OldCurrency>
+where
+	T: Config,
+	OldCurrency: 'static + ReservableCurrency<<T as frame_system::Config>::AccountId>,
+	BalanceOf<T>: From<OldCurrency::Balance>,
+{
+	const VERSION: u16 = 14;
+
+	fn max_step_weight() -> Weight {
+		T::WeightInfo::v14_migration_step()
+	}
+
+	fn step(&mut self) -> (IsFinished, Weight) {
+		let mut iter = if let Some(last_hash) = self.last_code_hash.take() {
+			old::CodeInfoOf::<T, OldCurrency>::iter_from(
+				old::CodeInfoOf::<T, OldCurrency>::hashed_key_for(last_hash),
+			)
+		} else {
+			old::CodeInfoOf::<T, OldCurrency>::iter()
+		};
+
+		if let Some((hash, code_info)) = iter.next() {
+			log::debug!(target: LOG_TARGET, "Migrating storage deposit for 0x{:?}", HexDisplay::from(&code_info.owner.encode()));
+
+			let remaining = OldCurrency::unreserve(&code_info.owner, code_info.deposit);
+
+			if remaining > Zero::zero() {
+				log::warn!(
+					target: LOG_TARGET,
+					"Code owner's account 0x{:?} for code {:?} has some non-unreservable deposit {:?} from a total of {:?} that will remain in reserved.",
+					HexDisplay::from(&code_info.owner.encode()),
+					hash,
+					remaining,
+					code_info.deposit
+				);
+			}
+
+			let unreserved = code_info.deposit.saturating_sub(remaining);
+			let amount = BalanceOf::<T>::from(unreserved);
+
+			T::Currency::hold(
+				&HoldReason::CodeUploadDepositReserve.into(),
+				&code_info.owner,
+				amount,
+			)
+			.map(|_| {
+				log::debug!(
+					target: LOG_TARGET,
+					"{:?} held on the code owner's account 0x{:?} for code {:?}.",
+					amount,
+					HexDisplay::from(&code_info.owner.encode()),
+					hash,
+				);
```

Comment:
nit, I would put that before `T::Currency::hold`, maybe changing held to Holding, to avoid adding this extra map